sdks/java/core/build.gradle

```diff
@@ -69,6 +69,7 @@ dependencies {
   compile library.java.protobuf_java
   compile library.java.commons_compress
   compile library.java.commons_lang3
+  compile library.java.jsr305
```

Comment:
You cannot do this due to the dependency license Notice that Beam moved to spotbugs for the missing javax.annotations (but it still has the same licensing issue).
https://github.com/apache/beam/blob/8066d78f0fd2237b718859d4a776511203880df0/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L738-L741

Reply:
For extra ref the licenses we cannot include in Apache Software Foundations (ASF) projects is known as the [Category X dependencies](https://www.apache.org/legal/resolved.html#category-x).

Reply:
Responded to you in JIRA ticket.

https://issues.apache.org/jira/browse/BEAM-8917?focusedCommentId=16991659&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16991659

Reply:
Good info thanks! We probably still would need to go with spotbugs anyway to be consistent with the previous upgrade, but maybe it is worth to focus on the big issue that's on https://issues.apache.org/jira/browse/BEAM-8858

Reply:
Sure, I can wait for [BEAM-8858](https://issues.apache.org/jira/browse/BEAM-8858) cleared.

What is "the previous upgrade" you're referring?

Reply:
We have a special configuration for annotation-only deps here: https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L741

Reply:
Sorry @suztomo  I was not clear, I referred to spotbugs-annotations package the one on Kenn's link

Reply:
@kennknowles @iemejia Thank you for clarificaiton. I added an explanation at the Kenn's link https://github.com/apache/beam/pull/10324/files#diff-23833058cbf2c1172b90e7764032aa59R732 .

@iemejia Do you think this PR solves the missing Nullable class you observed in [BEAM-8858](https://issues.apache.org/jira/browse/BEAM-8858)?

Reply:
I think it covers at least the legal part, it feels weird we did not catch this before. The code that uses programatically the annotation is more than 1y old already.
https://github.com/apache/beam/commit/788ce61bd7c48bc16a8eaa93a46ac403155f4422#diff-504e2e0131eda09163b086becec92f3cR91
I wonder if the vendoring part was the reason this became explicit. @kennknowles what do you think, I am inclined to merge this as a quickfix but with some doubts because of leaking the supposed to be  'annotation' only dep now into core.

Reply:
The page & the pom say it is ASL 2.0: https://mvnrepository.com/artifact/com.google.code.findbugs/jsr305/3.0.2

So it is fine, but the comment in BeamModulePlugin says BSD.

Reply:
Can you add it to the list of `compileOnlyAnnotationDeps` where you see `spotbugs-annotations`?

I am actually sort of surprised this works, since spotbugs operates on bytecode, not during compilation. I guess the annotations are retained and inspected in a way that does not require having the annotation library present. I have heard that any annotation with a class name (insensitive) of "nullable" will work, for example.

Noting in yet another place that spotbugs nullability analysis appears to be nonfunctional anyhow.

Reply:
In https://github.com/findbugsproject/findbugs/issues/128#issuecomment-310932778, iloveeclipse confirmed that JSR305 has BSD license and was unexpectedly uploaded to Maven Central with a wrong license.

I moved the JSR305 explanation closer to  spotbugs-annotations.

Reply:
The issue is that it is not a `compileOnlyAnnotation` anymore because it is used to resolve nullability of Schema based fields per https://github.com/apache/beam/commit/788ce61bd7c48bc16a8eaa93a46ac403155f4422#diff-504e2e0131eda09163b086becec92f3cR91 so it is needed at runtime too, or am I misreading this?

Reply:
That explanation is correct.

Reply:
Makes sense. The LGPL annotations are spotbugs-annotations which have things like SuppresFBWarnings and a few other utility annotations. jsr305 is fine. The landscape of these annotations and what they do is actually quite strange, incomplete, and also changed.